@uportal/user-profile-menu/src/components/ProfileMenu.vue

```diff
@@ -0,0 +1,217 @@
+      <div
+        class="avatar"
+        :style="{
+          width: avatarSize,
+          height: avatarSize,
+          'font-size': avatarSize
+        }"
+      >
+        <img :src="avatarUrl" v-if="avatarUrl && !useInitials" />
+        <span class="fallback" v-else>{{ fallbackText }}</span>
+      </div>
+      <span>{{ displayText }}</span>
+    </button>
+    <div class="profile-dropdown" v-if="menuOpen">
+      <div class="profile-dropdown-section profile-dropdown-header">
+        <slot name="header"></slot>
+      </div>
+      <div class="profile-dropdown-section profile-dropdown-content">
+        <slot name="default"></slot>
+      </div>
+      <div class="profile-dropdown-section profile-dropdown-footer">
+        <slot name="footer"></slot>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import oidc from '@uportal/open-id-connect';
+
+export default {
+  name: 'profile',
+  data() {
+    return {
+      menuOpen: false,
+      user: {
+        name: '',
+        family_name: '',
+        given_name: '',
+        image: null
+      },
+      dataLoaded: false,
+      hasError: false,
+      errorMessage: ''
+    };
+  },
+  computed: {
+    avatarUrl: function() {
+      return this.user.image;
+    },
+    displayText: function() {
+      return this.user.name;
+    },
+    fallbackText: function() {
+      return (
+        this.user.family_name.substr(0, 1) + this.user.given_name.substr(0, 1)
+      );
+    }
+  },
+  props: {
+    avatarSize: { type: String, default: '50px' },
+    useInitials: { type: Boolean, default: false, required: false },
+    oidcUrl: {
+      type: String,
+      default: '/uPortal/api/v5-1/userinfo'
+    }
+  },
+  methods: {
+    toggleMenu() {
+      this.menuOpen = !this.menuOpen;
+    },
+    handleOutsideClick(event) {
+      const menu = this.$refs.userProfileMenu;
+      const target = event.target;
+      const shadow = target.shadowRoot ? target.shadowRoot : false;
+      if (menu !== target && !shadow && !menu.contains(target)) {
+        this.menuOpen = false;
+      }
+    },
+    handleOidcError() {
+      this.hasError = true;
+      this.errorMessage = 'There was a problem authorizing this request.';
+    },
+    async getToken() {
+      try {
+        return await oidc({ userInfoApiUrl: this.oidcUrl });
+      } catch (err) {
+        this.handleOidcError(err);
+      }
+    },
+    async fetchMenuData() {
+      const token = this.debug ? null : (await this.getToken()).decoded;
+      this.user = token;
+    }
+  },
+  mounted() {
+    document.addEventListener('click', this.handleOutsideClick, false);
+    this.fetchMenuData();
+  }
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style lang="scss">
+.profile-menu-container {
+  color: white;
+  display: flex;
+  margin-left: auto;
+  position: relative;
+  padding: 0;
+
+  a {
+    text-decoration: none;
+    color: var(--user-prof-menu-fg-color, #495057);
```

Comment:
Fallbacks for browsers that don't support CSS Variables could be a nice touch :sparkles: 

```suggestion
    color: #495057;
    color: var(--user-prof-menu-fg-color, #495057);
```